### ISSUE TITLE
What did "offsets" and "positions" mean in yaml?

### ISSUE THREAD

**Issue by bxc237** [MarkNaeem/ros_controllers#6]
Hi, thank you for sharing your great work. Could you explain more about the "offsets" and "positions" parameters? Thank you!

**Comment by MarkNaeem** [comment]
Thanks for your interest in my work!
Basically, the position is where your wheel is located w.r.t the base frame. Offset is how far your wheel is away from its centre of rotation (described by the position). Some wheels have zero offsets, that is, they rotate around their centre of rotation directly. This works for vehicles with independent steering motors per wheel, where they can be mounted directly above the wheel. In the case of something like Ackermann-steering vehicles (like most of our cars nowadays), it's impossible to make the wheel turn around its centre of rotation and there is always a little offset.

**Comment by bxc237** [comment]
Thank you for your explanation! Close this issue. I also found some bugs in the code.  All those `velocities/positions_previous` vectors have not been `resize` in `init()`, it will cause problem if `publish_wheel_joint_controller_state` set to `true`...Sometimes, the controller would ignore the joint limits of holder joints, it will command the joint to `-3.14/3.14` instead of `0` when I set `limits` to `[-2,2]`...I will dig deeper into the code. Thank you.
